17/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/block-editor/block-grid-editor.md

```diff
@@ -577,60 +703,39 @@ public class BlockGridLayout
 // this represents an item in the block grid layout collection
 public class BlockGridLayoutItem
 {
-    public BlockGridLayoutItem(Udi contentUdi, Udi settingsUdi, int columnSpan, int rowSpan)
+    public BlockGridLayoutItem(int columnSpan, int rowSpan, Guid contentKey, Guid settingsKey)
     {
-        ContentUdi = contentUdi;
-        SettingsUdi = settingsUdi;
         ColumnSpan = columnSpan;
         RowSpan = rowSpan;
+        ContentKey = contentKey;
+        SettingsKey = settingsKey;
     }
 
-    [JsonPropertyName("contentUdi")]
-    public Udi ContentUdi { get; }
-
-    [JsonPropertyName("settingsUdi")]
-    public Udi SettingsUdi { get; }
-
-    [JsonPropertyName("areas")]
-    // areas are omitted from this sample for abbreviation
-    public object[] Areas { get; } = { };
-
     [JsonPropertyName("columnSpan")]
     public int ColumnSpan { get; }
 
     [JsonPropertyName("rowSpan")]
     public int RowSpan { get; }
 
-}
+    [JsonPropertyName("contentKey")]
+    public Guid ContentKey { get; }
 
-// this represents an item in the block grid content or settings data collection
-public class BlockGridElementData
-{
-    public BlockGridElementData(Guid contentTypeKey, Udi udi)
-    {
-        ContentTypeKey = contentTypeKey;
-        Udi = udi;
-    }
+    [JsonPropertyName("settingsKey")]
+    public Guid SettingsKey { get; }
 
-    [JsonPropertyName("contentTypeKey")]
-    public Guid ContentTypeKey { get; }
-
-    [JsonPropertyName("udi")]
-    public Udi Udi { get; }
-
-    [JsonExtensionData]
-    public Dictionary<string, object>? Data { get; set;}
+    [JsonPropertyName("areas")]
+    // areas are omitted from this sample for abbreviation
+    public object[] Areas { get; } = [];
 }
 ```
 {% endcode %}
 
-9. By injecting [ContentService](https://apidocs.umbraco.com/v15/csharp/api/Umbraco.Cms.Core.Services.ContentService.html) and [ContentTypeService](https://apidocs.umbraco.com/v15/csharp/api/Umbraco.Cms.Core.Services.ContentTypeService.html) into an API controller, we can transform the raw data into Block Grid JSON. It can then be saved to the target content item. Create a class called **BlockGridTestController.cs** containing the following:
+8.  Create a class called **BlockGridTestController.cs**. By injecting `IContentService` and `IContentTypeService` into an API controller, we can transform the raw data into Block Grid JSON. It can then be saved to the target content item.
```

Comment:
```suggestion
8.  Create a class called **BlockGridTestController.cs**. By injecting `IContentService` and `IContentTypeService` into an API controller, the raw data can be transformed into Block Grid JSON. It can then be saved to the target content item.
```